docs/user-guide/kubectl-conventions.md

```diff
@@ -29,6 +29,40 @@ In order for `kubectl run` to satisfy infrastructure as code:
 * If features are needed that are not expressible via `kubectl run` flags, switch to configuration files checked into source control.
 * Pin to a specific generator version, such as `kubectl run --generator=deployment/v1beta1`
 
+#### Generators
+
+`kubectl run` allows you to generate following resources (using `--generator` flag):
+
+* Pod - use `run-pod/v1`.
+* Replication controller - use `run/v1`.
+* Deployment - use `deployment/v1beta1`.
+* Job (using `extension/v1beta1` endpoint) - use `job/v1beta1`.
+* Job - use `job/v1`.
+* ScheduledJob - use `scheduledjob/v2alpha1`.
+
+Additionally, if no generator flag was specified other flags will suggest using
+a specific generator.  Below table shows which flags force using specific generators,
+depending on your cluster version:
+
+|   Generated Resource   | Cluster v1.4          | Cluster v1.3          | Cluster v1.2                               | Cluster v1.1 and eariler                   |
+|:----------------------:|-----------------------|-----------------------|--------------------------------------------|--------------------------------------------|
+| Pod                    | `--restart=Never`     | `--restart=Never`     | `--generator=run-pod/v1`                   | `--restart=OnFailure` OR `--restart=Never` |
+| Replication Controller | `--generator=run/v1`  | `--generator=run/v1`  | `--generator=run/v1`                       | `--restart=Always`                         |
+| Deployment             | `--restart=Always`    | `--restart=Always`    | `--restart=Always`                         | N/A                                        |
+| Job                    | `--restart=OnFailure` | `--restart=OnFailure` | `--restart=OnFailure` OR `--restart=Never` | N/A                                        |
+| Scheduled Job          | `--schedule=<cron>`   | N/A                   | N/A                                        | N/A                                        |
+
+It's important to note, that these flags will default the generator only when none
+is chosen.  This also means that combining any of above flags with `--generator` will
+not change the generator at any point.
```

Comment:
"This also means that combining any of above flags with `--generator` will not change the generator at any point." --> This is a bit ambiguous, since `--generator` is also in the table. Maybe rephrase this to something like:

[...] combining `--generator` with other flags won't change the `generator` you specified. 

For example, in a 1.4 cluster, if you specify `--restart=Always`, a Deployment will be created; if you specify `--restart=Always` and `--generator=run/v1`, a Replication Controller will be created instead. This becomes handy if you want to pin to a specific behavior with the generator, even when the defaulted generator is change in the future.